Bubble sort to main

SamroodAli opened the pull request:
Please review our code
We have done the bubble sort exercise for sorting both integers and strings using Ruby

lekegitrepo reviewed (changes requested):
## Status: Changes Required! :recycle:
### Hi @SamroodAli and @Stephanie041996 :clap: :clap:
#### You have done a good job :muscle: However, some requirements are missing/not implemented correctly. check my comments below :point_down:

#### 1) Readme:
 - Add instruction on how to clone and run the program locally

#### 2) implementation:
 - bubble_sort_by should raise an error(`LocalJumpError`) if a block is not given

#### _Reviewed by Temitayo Adeleke_

BrianSammit commented on bubble_sort.rb:
```diff
@@ -0,0 +1,40 @@
+  end
+end
+
+# test
+bubble_sort_words(['hi', 'hello', 'hello world', 'a']) do |left, right|
+  left.length - right.length
+end
+p bubble_sort([1, 6, 3, 7, 9])
```
From line 36 to line 40 arr debugging code. You use this code to check the code, those lines are debugging code. Please remove it or keep it in another branch. :hammer_and_wrench:

BrianSammit reviewed (changes requested):
CHANGES REQUIRED :red_circle: :recycle: 
---------------
Hello, @SamroodAli @Stephanie041996 you are doing an amazing job :muscle: :100: Your project looks really nice, I like the good structure of the code. :smile: 
However, there are a few changes to make to complete this project.  :hammer_and_wrench: see them below :eyes: :point_down: 
------------

- [x] Improve your Readme
- Write the concept of bubble sort, how it works.
- Improve your steps on how to clone the repository. Write a step-by-step as if you were someone that doesn't know anything about programming. 
- [OPTIONAL] put an image of the code. 

- [x] Delete your debugging code. remember that no debugging instructions in committed code. See my inline comments to know what I'm talking about.  :eyes: 
- [x] Your code should raise an error(LocalJumpError) if a block is not given. Just remove the if statement and it will work.  
- [ ] Improve your commit messages. Remember it must be imperative. [Here ](https://github.com/trein/dev-best-practices/wiki/Git-Commit-Best-Practices#formatting-rules) you can read about the formatting rules. [Here](https://docs.github.com/en/github/committing-changes-to-your-project/changing-a-commit-message) you can read about how to change a commit message. Keep these formatting rules in mind for the next commits. 
--------------
Make the required changes and send it again for another review :+1: 
Reachme on SLACK if you have any questions. 
@ Brian Sammit Cruz Rodriguez
HAPPY CODING!! :smile: 

![200 (1)](https://user-images.githubusercontent.com/64173189/111684497-c746ad80-87f4-11eb-9452-92e4bbe49efe.gif)

SamroodAli commented:
Thanks for the review
We have improved the readme file
updated commits, they are now in imperative present tense style
Added image of code, setting up instructions as well info about bubble sort in readme
Removed development test code in bubble sort file

yocosaka reviewed (changes requested):
##  Status: Changes Required♻️ 

**Great job so far, team**👍  

You two have made an amazing job🚀
There is only one point to improve before you go ahead.
Let's cover a few details together to make your project better😊 

### Required changes: 
- [ ] bubble_sort_by should raise an error if block is not given
Now it returns the same result as the result if a block is given.


Please request another review once you are ready👍

### With all my respect 🍀

Yoko Saka

andresporras3423 reviewed (changes requested):
### Status: Changes Required :recycle:
**Hello Sam Ali & Stephanie Sakuhuni** 
Good job with the improvements to this project :muscle:  :muscle:
your functions passes all the tests :clap: :clap: :clap:  
There is just one last detail to improve before you can go ahead:

- [ ] Replace the name of function "bubble_sort_words" with "bubble_sort_by"
 
Please request another review once you are ready! :+1:

BrianSammit reviewed (approved):
STATUS: APPROVED 🟢 
-------
Hello, again team!! you work really hard on this project 💪 great job working on the required changes. Your project is working perfectly 💯 🥇 
---------
You can merge this branch now 💯 
HAPPY CODING!! AND GOOD LUCK WITH THE COMMING PROJECTS. 😄 
![tenor (1)](https://user-images.githubusercontent.com/64173189/111808742-4a720d00-88a2-11eb-9345-6099d5f227ca.gif)
